### PR TITLE
Support rendering some media downloads as inline

### PR DESCRIPTION
**Pull request by Half-Shot**
This allows a few select basic text formats to be rendered inline in the browser rather than downloaded. This is primarily to support the established "pastebin" behaviour of some bridges where text files would be uploaded to represent longer messages.

The sytests that covered this have been removed. The complement tests have been modified in https://github.com/matrix-org/complement/pull/646.

### CODE REVIEW

**Review by erikjohnston — APPROVED** [review]

**Comment by erikjohnston** [comment]
@Half-Shot Ah, we'll need to update the sytests tests as well. The tests can be found in: https://github.com/matrix-org/sytest/tree/develop/tests/51media

**Comment by reivilibre** [comment]
related issue: https://github.com/matrix-org/synapse/issues/15885

File: synapse/media/_base.py [hunk 1]
```diff
@@ -50,6 +50,24 @@
     "text/xml",
 ]
 
+# A list of all content types that are "safe" to be rendered inline in a browser.
```
Comment (clokep) [inline synapse/media/_base.py:53]: See https://github.com/matrix-org/matrix-react-sdk/blob/a70fcfd0bcf7f8c85986da18001ea11597989a7c/src/utils/blobs.ts for what Element Web will attempt to render inline.

File: synapse/media/_base.py [hunk 2]
```diff
@@ -50,6 +50,24 @@
...
+    # We allow images listed under the common image list
+    # https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#common_image_file_types
+    # except SVGs
+    "image/apng",
+    "image/avif",
+    "image/gif",
+    "image/jpeg",
+    "image/png",
+    "image/webp",
+]
```
Comment (itsrachelfish) [inline synapse/media/_base.py:69]: @Half-Shot Is there a reason why video and audio content types aren't being included in this PR? :thinking:

Comment (josteink) [reply synapse/media/_base.py:69]: Seconding this comment. I was also wondering about this.

Comment (zopieux) [reply synapse/media/_base.py:69]: Glowing Bear, for instance, has been showing inline video and audio (using native HTML widgets) for years.

Comment (Half-Shot) [reply synapse/media/_base.py:84]: The list is now amended to follow what the matrix-react-sdk does.

**Comment by clokep** [comment]
I made a minor tweak to this -- to server the filename for inline content dispositions also. Put it back up for review.

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**